### PR TITLE
AssignmentTask Controller implementation#17

### PR DESCRIPTION
**Pull request by fedyafedko**

### CODE REVIEW

File: StudyHub/StudyHub.Common/DTO/AssignmentTaskOption/AssignmentTaskOptionDTO.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿namespace StudyHub.Common.DTO.AssignmentTaskOption;
+public class AssignmentTaskOptionDTO
+{
+    public string Label { get; set; }
```
Comment (Podgore) [inline StudyHub/StudyHub.Common/DTO/AssignmentTaskOption/AssignmentTaskOptionDTO.cs:4]: fix warnings

File: StudyHub/StudyHub/Controllers/AssignmentTaskController.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+{
+    private readonly IAssignmentTaskService _serviceTask;
+
+    public AssignmentTaskController(IAssignmentTaskService serviceTask)
+    {
+        _serviceTask = serviceTask;
+    }
+
+    [HttpPost]
+    public async Task<IActionResult> InsretTask(CreateAssignmentTaskDTO dto)
```
Comment (Podgore) [inline StudyHub/StudyHub/Controllers/AssignmentTaskController.cs:19]: fix naming

File: StudyHub/StudyHub.BLL/Services/AssignmentTaskService.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    public async Task<AssignmentTaskDTO> AddTask(CreateAssignmentTaskDTO dto)
+    {   
+        if (dto.Options.Count == 0)
+        {
+            throw new Exception("No options");
+        }
+
+        var entity = _mapper.Map<AssignmentTask>(dto);
+
+        if (_repositoryAssignment.FirstOrDefault(option => option.Id == dto.AssignmentId) == null)
```
Comment (aquaprogit) [inline StudyHub/StudyHub.BLL/Services/AssignmentTaskService.cs:36]:
```suggestion
        if (await _repositoryAssignment.FirstOrDefaultAsync(assignment => assignment.Id == dto.AssignmentId) == null)
```

File: StudyHub/StudyHub.BLL/Services/OptionsService.cs [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                taskOption,
+                _openEndedOptionRepository);
+        else if (taskOption.All(option => option.IsCorrect != null))
+            return await SeparationOptions(assignmentTaskId,
+                taskOption,
+                _choiceOptionRepository);
+        else
+        {
+
+            var nullOptions = taskOption
```
Comment (aquaprogit) [inline StudyHub/StudyHub.BLL/Services/OptionsService.cs:38]: what is purpose of this?

File: StudyHub/StudyHub.Common/DTO/AssignmentTask/CreateAssignmentTaskDTO.cs [hunk 5]
```diff
@@ -0,0 +1,11 @@
+﻿using StudyHub.Common.DTO.AssignmentTaskOption;
+
+namespace StudyHub.Common.DTO.AssignmentTask;
+
+public class CreateAssignmentTaskDTO
+{
+    public Guid AssignmentId { get; set; }
+    public string Label { get; set; } = string.Empty;
+    public int Mark { get; set; }
+    public List<AssignmentTaskOptionDTO> Options { get; set; } = null!;
```
Comment (aquaprogit) [inline StudyHub/StudyHub.Common/DTO/AssignmentTask/CreateAssignmentTaskDTO.cs:10]: consistency missing

File: StudyHub/StudyHub.Common/DTO/AssignmentTask/AssignmentTaskDTO.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+namespace StudyHub.Common.DTO.AssignmentTask;
+
+public class AssignmentTaskDTO
+{
+    public Guid Id { get; set; }
+    public Guid AssignmentId { get; set; }
+    public string Label { get; set; } = string.Empty;
+    public int Mark { get; set; }
+    public List<AssignmentTaskOptionDTO> Options { get; set; } = new List<AssignmentTaskOptionDTO>();
```
Comment (aquaprogit) [inline StudyHub/StudyHub.Common/DTO/AssignmentTask/AssignmentTaskDTO.cs:11]: consistency missing

File: StudyHub/StudyHub.DAL/Repositories/Repository.cs [hunk 7]
```diff
@@ -10,7 +9,7 @@ public class Repository<TEntity> : IRepository<TEntity>
     where TEntity : class
 {
     private readonly ApplicationDbContext _context;
-    private readonly DbSet<TEntity> _table;
+    private readonly DbSet<TEntity> _table;
```
Comment (aquaprogit) [inline StudyHub/StudyHub.DAL/Repositories/Repository.cs:12]: why do you need empty space here?

File: StudyHub/StudyHub/Controllers/AssignmentTaskController.cs (hunk 2 above)
Comment (aquaprogit) [inline StudyHub/StudyHub/Controllers/AssignmentTaskController.cs:19]:
```suggestion
    public async Task<IActionResult> InsertTask(CreateAssignmentTaskDTO dto)
```

File: StudyHub/StudyHub/Program.cs [hunk 8]
```diff
@@ -2,9 +2,15 @@
...
 using StudyHub.DAL.Repositories.Interfaces;
 using StudyHub.DAL.Repositories;
+using StudyHub.BLL.Profiles;
+using StudyHub.BLL.Interfaces;
+using StudyHub.BLL.Services;
 
 var builder = WebApplication.CreateBuilder(args);
 
+builder.Services.AddAutoMapper(typeof(AssignmentTaskProfile).Assembly);
+builder.Services.AddAutoMapper(typeof(OptionsProfile).Assembly);
```
Comment (aquaprogit) [inline StudyHub/StudyHub/Program.cs:12]: I don't think that you need to add mapper for every profile
Check documentation about this

**Review by aquaprogit — CHANGES_REQUESTED** [review]

**Review by aquaprogit — CHANGES_REQUESTED** [review]

**Review by aquaprogit — DISMISSED** [review]

**Review by aquaprogit — APPROVED** [review]

**Review by Podgore — APPROVED** [review]

**Review by Podgore — APPROVED** [review]

**Pull request merged**